File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Persistencia_de_Objetos/Ejercicio3/Python/main.py

from Cliente import Cliente
from ArchivoCliente import ArchivoCliente

if __name__ == "__main__":
    nombre_archivo_clientes = "clientes_ejercicio3.txt"
    archivo_cliente = ArchivoCliente(nombre_archivo_clientes)

    archivo_cliente.limpiar_archivo()

    print("\n--- Guardando Clientes ---")
    cli1 = Cliente(101, "Ana Lopez", 77712345)
    cli2 = Cliente(102, "Juan Perez", 70098765)
    cli3 = Cliente(103, "Maria Gomez", 61234567)
    cli4 = Cliente(104, "Pedro Sanchez", 77712345) 

    archivo_cliente.guarda_cliente(cli1)
    archivo_cliente.guarda_cliente(cli2)
    archivo_cliente.guarda_cliente(cli3)
    archivo_cliente.guarda_cliente(cli4)

    print("\n--- Buscando Cliente por ID ---")
    # b) 
    id_busqueda1 = 102
    cliente_encontrado_id1 = archivo_cliente.buscar_cliente(id_busqueda1)
    if cliente_encontrado_id1:
        print(f"Cliente con ID {id_busqueda1} encontrado: {cliente_encontrado_id1}")
    else:
        print(f"Cliente con ID {id_busqueda1} no encontrado.")

    id_busqueda2 = 105
    cliente_encontrado_id2 = archivo_cliente.buscar_cliente(id_busqueda2)
    if cliente_encontrado_id2:
        print(f"Cliente con ID {id_busqueda2} encontrado: {cliente_encontrado_id2}")
    else:
        print(f"Cliente con ID {id_busqueda2} no encontrado.")

    print("\n--- Buscando Cliente por Numero de Celular ---")
    # c) 
    telefono_busqueda1 = 77712345
    cliente_encontrado_tel1 = archivo_cliente.buscar_celular_cliente(telefono_busqueda1)
    if cliente_encontrado_tel1:
        
        print(f"Cliente con celular {telefono_busqueda1} encontrado: {cliente_encontrado_tel1}")
    else:
        print(f"Cliente con celular {telefono_busqueda1} no encontrado.")

    telefono_busqueda2 = 60000000
    cliente_encontrado_tel2 = archivo_cliente.buscar_celular_cliente(telefono_busqueda2)
    if cliente_encontrado_tel2:
        print(f"Cliente con celular {telefono_busqueda2} encontrado: {cliente_encontrado_tel2}")
    else:
        print(f"Cliente con celular {telefono_busqueda2} no encontrado.")

    
    print(f"\nContenido final del archivo '{nombre_archivo_clientes}':")
    try:
        with open(nombre_archivo_clientes, 'r') as f:
            print(f.read())
    except FileNotFoundError:
        print("El archivo no existe aún.")

File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Persistencia_de_Objetos/Ejercicio3/Java/Main.java
package Persistencia_de_Objetos.Ejercicio3.Java;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class Main {
    public static void main(String[] args) {
        String nombreArchivoClientes = "lientes_ejercicio3.txt";
        ArchivoCliente archivoCliente = new ArchivoCliente(nombreArchivoClientes);

        archivoCliente.limpiarArchivo();

        System.out.println("\n--- Guardando Clientes ---");
        Cliente cli1 = new Cliente(101, "Ana Lopez", 77712345);
        Cliente cli2 = new Cliente(102, "Juan Perez", 70098765);
        Cliente cli3 = new Cliente(103, "Maria Gomez", 61234567);
        Cliente cli4 = new Cliente(104, "Pedro Sanchez", 77712345); // Otro cliente con el mismo teléfono

        archivoCliente.guardaCliente(cli1);
        archivoCliente.guardaCliente(cli2);
        archivoCliente.guardaCliente(cli3);
        archivoCliente.guardaCliente(cli4);

        System.out.println("\n--- Buscando Cliente por ID ---");
        // b)
        int idBusqueda1 = 102;
        Cliente clienteEncontradoId1 = archivoCliente.buscarCliente(idBusqueda1);
        if (clienteEncontradoId1 != null) {
            System.out.println("Cliente con ID " + idBusqueda1 + " encontrado: " + clienteEncontradoId1);
        } else {
            System.out.println("Cliente con ID " + idBusqueda1 + " no encontrado.");
        }

        int idBusqueda2 = 105;
        Cliente clienteEncontradoId2 = archivoCliente.buscarCliente(idBusqueda2);
        if (clienteEncontradoId2 != null) {
            System.out.println("Cliente con ID " + idBusqueda2 + " encontrado: " + clienteEncontradoId2);
        } else {
            System.out.println("Cliente con ID " + idBusqueda2 + " no encontrado.");
        }

        System.out.println("\n--- Buscando Cliente por Numero de Celular ---");
        // c) 
        int telefonoBusqueda1 = 77712345;
        Cliente clienteEncontradoTel1 = archivoCliente.buscarCelularCliente(telefonoBusqueda1);
        if (clienteEncontradoTel1 != null) {
          
            System.out.println("Cliente con celular " + telefonoBusqueda1 + " encontrado: " + clienteEncontradoTel1);
        } else {
            System.out.println("Cliente con celular " + telefonoBusqueda1 + " no encontrado.");
        }

        int telefonoBusqueda2 = 60000000;
        Cliente clienteEncontradoTel2 = archivoCliente.buscarCelularCliente(telefonoBusqueda2);
        if (clienteEncontradoTel2 != null) {
            System.out.println("Cliente con celular " + telefonoBusqueda2 + " encontrado: " + clienteEncontradoTel2);
        } else {
            System.out.println("Cliente con celular " + telefonoBusqueda2 + " no encontrado.");
        }

        System.out.println("\nContenido final del archivo '" + nombreArchivoClientes + "':");
        try (BufferedReader reader = new BufferedReader(new FileReader(nombreArchivoClientes))) {
            String line;
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
            }
        } catch (IOException e) {
            System.err.println("Error al leer el archivo para mostrar su contenido final: " + e.getMessage());
        }
    }
}
